ddtrace/contrib/botocore/services/sqs.py

```diff
@@ -161,17 +161,23 @@ def patched_sqs_api_call(original_func, instance, args, kwargs, function_vars):
                     result = original_func(*args, **kwargs)
                     core.dispatch(f"botocore.{endpoint_name}.{operation}.post", [params, result])
 
-                core.dispatch("botocore.patched_sqs_api_call.success", [ctx, result, set_response_metadata_tags])
+                core.dispatch("botocore.patched_sqs_api_call.success", [ctx, result])
 
                 if func_run_err:
                     raise func_run_err
                 return result
             except botocore.exceptions.ClientError as e:
                 core.dispatch(
                     "botocore.patched_sqs_api_call.exception",
-                    [ctx, e.response, botocore.exceptions.ClientError, set_response_metadata_tags],
+                    [
+                        ctx,
+                        e.response,
+                        botocore.exceptions.ClientError,
+                        config.botocore.operations[ctx[ctx["call_key"]].resource].is_error_code,
+                    ],
                 )
                 raise
+        parent_ctx.end()
```

Comment:
do we need this in a finally?

or... outside of the `if should_instrument` block? would it cause us to have a leaky/unclosed context?

Reply:
same with the one above?

what happens if we never end an `ExecutionContext`?

Reply:
good catch. I think when I wrote it I was telling myself that it would get closed when it left scope, but thinking about it again I'm not so sure. Better to close explicitly with a `finally`.